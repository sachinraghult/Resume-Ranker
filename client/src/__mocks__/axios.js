export default {
  get: jest.fn(() => {
    Promise.resolve({ data: {} });
    console.log("1");
  }),
};
