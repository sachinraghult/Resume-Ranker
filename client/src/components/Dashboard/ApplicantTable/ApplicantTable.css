.widgetLg {
  position: relative;
  -webkit-box-shadow: 0px 0px 15px -10px rgba(0, 0, 0, 0.859);
  box-shadow: 0px 0px 15px -10px rgba(0, 0, 0, 0.75);
  padding: 50px;
  margin-top: 50px;
  width: 100%;
  margin-bottom: 25px;
  background-color: #f2f2f2;
}

.widgetLgTitle {
  font-size: 22px;
  font-weight: 600;
}

.widgetLgTable {
  width: 100%;
  border-spacing: 20px;
  /* border-width: 2px; */
}

.widgetLgTh {
  text-align: left;
}

.widgetLgUser {
  display: flex;
  align-items: center;
  font-weight: 600;
}

.widgetLgImg {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  object-fit: cover;
  margin-right: 10px;
}

.widgetLgDate,
.widgetLgAmount {
  font-weight: 300;
}

.widgetLgButton {
  padding: 5px 7px;
  border: none;
  border-radius: 10px;
}

.widgetLgButton.Approved {
  background-color: #e5faf2;
  color: #3bb077;
}
.widgetLgButton.Declined {
  background-color: #fff0f1;
  color: #d95087;
}
.widgetLgButton.Pending {
  background-color: #ebf1fe;
  color: #2a7ade;
}

th,
td {
  border: 1px solid black;
  margin: 0px 0px;
  padding: 5px 5px;
}

.widgetLg .cellAction .viewButton {
  padding: 2px 5px;
  border-radius: 5px;
  color: darkblue;
  border: 1px dotted rgba(0, 0, 139, 0.596);
  cursor: pointer;
}
