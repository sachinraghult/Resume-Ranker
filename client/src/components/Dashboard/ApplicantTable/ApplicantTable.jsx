import { useEffect, useState, useContext } from "react";
import axios from "axios";
import { Link, useLocation } from "react-router-dom";
import "./ApplicantTable.css";
import { DataGrid } from "@mui/x-data-grid";
import { Context } from "../../../context/Context";

import { SERVER_URL } from "../../../config";

export default function ApplicantTable() {
  const location = useLocation();
  const path = location.pathname.split("/")[2];
  const { authToken } = useContext(Context);
  const [row, setRow] = useState([]);
  const formatDate = (dateString) => {
    const options = { year: "numeric", month: "long", day: "numeric" };
    return new Date(dateString).toLocaleDateString(undefined, options);
  };

  useEffect(() => {
    const fetchAppTable = async () => {
      const res = await axios.get(
        `${SERVER_URL}/jobSpecificDashboard/table/table/` + path,
        {
          headers: { Authorization: authToken },
        }
      );
      if (res) {
        setRow(res.data);
      }
    };

    fetchAppTable();
  }, []);

  function onEmailClick(email) {
    window.open(`mailto:${email}`);
  }

  const columns = [
    // {
    //   field: "id",
    //   headerName: "S.NO",
    //   width: 100,
    //   renderCell: (params) => {
    //     return <div> {params.row.id + 1}</div>;
    //   },
    // },
    {
      field: "ApplicantName",
      headerName: "Applicant Name",
      width: 350,
      renderCell: (params) => {
        return (
          <div>
            <b>{params.row.ApplicantName}</b>
          </div>
        );
      },
    },
    {
      field: "mailID",
      headerName: "Applicant Detail",
      width: 250,
      renderCell: (params) => {
        return (
          <div className="cellAction">
            {params.row.mailID.split("&")[0] === "from bulk resumes" ? (
              <a
                href={`/ranking/${params.row.mailID.split("&")[1]}`}
                target="_blank"
                style={{ textDecoration: "none" }}
              >
                <div>Bulk Upload Resume</div>
              </a>
            ) : (
              <div
                style={{ color: "brown", cursor: "pointer" }}
                onClick={() => onEmailClick(params.row.mailID)}
              >
                {params.row.mailID}
              </div>
            )}
          </div>
        );
      },
    },
    {
      field: "Date",
      headerName: "Date",
      renderCell: (params) => {
        return <div> {formatDate(params.row.Date)}</div>;
      },
      sortable: false,
      width: 200,
    },
  ];

  const actionColumn = [
    {
      field: "action",
      headerName: "Resume",
      width: 200,
      renderCell: (params) => {
        return (
          <div className="cellAction">
            <a
              href={`https://drive.google.com/uc?id=${params.row.action}`}
              target="_blank"
              style={{ textDecoration: "none" }}
            >
              <div className="viewButton">View Resume</div>
            </a>
          </div>
        );
      },
    },
  ];

  console.log(row);
  return (
    <div className="widgetLg">
      <h3 className="widgetLgTitle">Recent Applications</h3>
      <br></br>

      <div style={{ height: 400, width: "100%", backgroundColor: "#e8e8e5" }}>
        <DataGrid
          rows={row}
          columns={columns.concat(actionColumn)}
          pageSize={5}
          rowsPerPageOptions={[5]}
          checkboxSelection
        />
      </div>
    </div>
  );
}
