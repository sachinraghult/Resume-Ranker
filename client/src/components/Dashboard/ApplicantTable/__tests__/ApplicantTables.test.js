import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import ApplicantTable from "../ApplicantTable";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";
import axios, { AxiosResponse } from "axios";

jest.mock("axios", () => ({
  ...jest.requireActual("axios"),
  get: jest.fn(),
}));

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <ApplicantTable />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const data = [
    {
      id: 0,
      ApplicantName: "User1",
      mailID: "u1@gmail.com",
      Date: "2022-07-10T18:41:22.865Z",
      action: "1O6qiBjctAGJ5TjrVnWCUtJRwkLaoKR6c",
    },
    {
      id: 1,
      ApplicantName: "User 2",
      mailID: "u2@gmail.com",
      Date: "2022-07-10T18:43:46.163Z",
      action: "1iHk55CSNpighYbC4dyfaD8rjHrJk5zWL",
    },
    {
      id: 2,
      ApplicantName: "User 3",
      mailID: "u3@gmail.com",
      Date: "2022-07-10T18:46:18.006Z",
      action: "15yc5-EmeAsrcz-1pbx2jEcjGosVwIupT",
    },
  ];
  axios.get.mockImplementationOnce(() => Promise.resolve({ data: data }));
  render(
    <BrowserRouter>
      <ApplicantTable />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ApplicantTable />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
