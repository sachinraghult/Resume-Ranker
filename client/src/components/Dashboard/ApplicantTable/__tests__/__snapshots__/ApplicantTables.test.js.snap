// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="widgetLg"
>
  <h3
    className="widgetLgTitle"
  >
    Recent Applications
  </h3>
  <br />
  <div
    style={
      Object {
        "backgroundColor": "#e8e8e5",
        "height": 400,
        "width": "100%",
      }
    }
  >
    <div
      aria-colcount={5}
      aria-multiselectable={false}
      aria-rowcount={0}
      className="MuiDataGrid-root MuiDataGrid-root--densityStandard css-r11z79-MuiDataGrid-root"
      role="grid"
    >
      <div
        className="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          className="MuiDataGrid-overlay css-18rq0oa-MuiDataGrid-overlay"
          error={[TypeError: Cannot set property 'scrollLeft' of null]}
        >
          An error occurred.
        </div>
      </div>
    </div>
  </div>
</div>
`;
